aten/src/ATen/native/cpu/PowKernel.cpp

```diff
@@ -0,0 +1,131 @@
+#include <cmath>
+#include <ATen/Dispatch.h>
+#include <ATen/Parallel.h>
+#include <ATen/cpu/vec256/vec256.h>
+#include <ATen/native/TensorIterator.h>
+#include <ATen/native/Pow.h>
+#include <ATen/native/cpu/Loops.h>
+
+namespace at { namespace native {
+
+namespace {
+
+void pow_tensor_tensor_kernel(TensorIterator& iter) {
+  if (isFloatingType(iter.dtype())) {
+    AT_DISPATCH_FLOATING_TYPES(iter.dtype(), "pow", [&]() {
+      using Vec = Vec256<scalar_t>;
+      cpu_kernel_vec(iter,
+        [=](scalar_t self, scalar_t exp) -> scalar_t {
+          return std::pow(self, exp);
+        },
+        [&](Vec self, Vec exp) -> Vec {
+          return self.pow(exp);
+        }
+      );
+    });
+  } else {
+    AT_DISPATCH_INTEGRAL_TYPES(iter.dtype(), "pow", [&]() {
+      cpu_kernel(iter,
+        [=](scalar_t self, scalar_t exp) -> scalar_t {
+          return std::pow(self, exp);
+        }
+      );
+    });
+  }
+}
+
+void pow_tensor_scalar_kernel(TensorIterator& iter, Scalar exp_scalar) {
+  if (isFloatingType(iter.dtype())) {
+    AT_DISPATCH_FLOATING_TYPES(iter.dtype(), "pow", [&]() {
+      using Vec = Vec256<scalar_t>;
+      auto exp = exp_scalar.to<scalar_t>();
+      if (exp == 0.5) {
+        cpu_kernel_vec(iter,
+          [](scalar_t self) -> scalar_t { return std::sqrt(self); },
+          [](Vec self) -> Vec { return self.sqrt(); }
+        );
+      } else if (exp == 2) {
+        cpu_kernel_vec(iter,
+          [](scalar_t self) -> scalar_t { return self * self; },
+          [](Vec self) -> Vec { return self * self; }
+        );
+      } else if (exp == 3) {
+        cpu_kernel_vec(iter,
+          [](scalar_t self) -> scalar_t { return self * self * self; },
+          [](Vec self) -> Vec { return self * self * self; }
+        );
+      } else if (exp == -0.5) {
+        cpu_kernel_vec(iter,
+          [](scalar_t self) -> scalar_t { return 1.0 / std::sqrt(self); },
+          [](Vec self) -> Vec { return self.sqrt().reciprocal(); }
+        );
+      } else if (exp == -1) {
+        cpu_kernel_vec(iter,
+          [](scalar_t self) -> scalar_t { return 1.0 / self; },
+          [](Vec self) -> Vec { return self.reciprocal(); }
+        );
+      } else if (exp == -2) {
+        cpu_kernel_vec(iter,
+          [](scalar_t self) -> scalar_t { return 1.0 / (self * self); },
+          [](Vec self) -> Vec { return (self * self).reciprocal(); }
+        );
+      } else {
+        cpu_kernel_vec(iter,
+          [=](scalar_t self) -> scalar_t { return std::pow(self, exp); },
+          [=](Vec self) -> Vec { return self.pow(exp); }
+        );
+      }
+    });
+  } else {
+    AT_DISPATCH_INTEGRAL_TYPES(iter.dtype(), "pow", [&]() {
+      using Vec = Vec256<scalar_t>;
+      auto exp = exp_scalar.to<scalar_t>();
+      if (exp == 0.5) {
```

Comment:
I'm afraid this might change current pow() functionality:

```python
In [65]: x = torch.empty(4, dtype=torch.long)                                                                                                         

In [66]: x.fill_(4)                                                                                                                                   
Out[66]: tensor([4, 4, 4, 4])

In [67]: x.pow(0.5)                                                                                                                                   
Out[67]: tensor([1, 1, 1, 1]) # Suddenly!!!! (on master)
```